crates/glue/src/types.rs

```diff
@@ -1268,8 +1266,13 @@ fn add_function_type<'a>(
 
     let name = format!("RocFunction_{closure_var:?}");
 
-    let id = env.lambda_set_ids.get(&closure_var).unwrap();
-    let extern_name = format!("roc__mainForHost_{}_caller", id.0);
+    let extern_name = match env.lambda_set_ids.get(&closure_var) {
+        Some(id) => format!("roc__mainForHost_{}_caller", id.0),
+        None => {
+            debug_assert!(is_toplevel);
+            String::from("this_extern_should_not_be_used_this_is_a_bug")
+        }
+    };
```

Comment:
I think we should ultimately refactor so that this hackery is not required. e.g. make a distinction for glue between (lifted-to-the) top-level functions and closures